src/java/org/apache/cassandra/index/sai/disk/PostingListRangeIterator.java

```diff
@@ -109,7 +116,10 @@ protected PrimaryKey computeNext()
             if (rowId == PostingList.END_OF_STREAM)
                 return endOfData();
 
-            return primaryKeyMap.primaryKeyFromRowId(rowId);
+            PrimaryKey pk = primaryKeyMap.primaryKeyFromRowId(rowId);
+            if (ssTableRowIdToScoreMap != null)
+                queryContext.recordScore(pk, ssTableRowIdToScoreMap.get(rowId));
```

Comment:
if `rowId` doesn't exist in `ssTableRowIdToScoreMap`, it returns 0.0 which will be recorded in `recordScore`. The score will be wrong

Reply:
Great point. I'll address this. Another issue with the PR is that it effectively only caches scores that are positive, but different similarity functions do support negative values, so we should revisit that as well.

Reply:
Not so, all similarities are given in a range of 0 .. 1

Reply:
Perfect, that's much easier to work with.